### ISSUE TITLE
@Viewscoped beans are not replicated in cluster environment

### ISSUE THREAD

**Issue by hn310** [payara/Payara#2739]
### 1. Cluster structure
Load balancer -> web server -> 2 payara instances (so called Node 1 and Node 2)
### 2. Problem
When I opened A.xhtml, first request went to Node 1. @Viewscoped bean was created. 
I turned off Node 1, then click a button on A.xhtml to reload the view, request went to Node 2.
However, @Viewscoped bean datas were lost on Node 2.

I simply sent a POST request to reload the page (which means I'm still at the same view), so I expect the datas in @Viewscoped bean would still be kept.
After I click the button, if the request goes to the same instance, the datas will be kept as expected.
If the request goes to the other instance, the datas will be lost.

Note that all my @Sessionscoped beans were correctly replicated and their datas were perfectly fine, so I don't think there is any problem with my session replication.
Is this normal behavior?

### 3. Technology
- Payara Server 4.1.1.163
- JSF 2.2 (Mojarra implementation)
- Using Hazelcast for session replication
- JDK: 1.8.0_102

### 4. Detailed configuration
4.1. glassfish-web-xml
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE glassfish-web-app PUBLIC "-//GlassFish.org//DTD GlassFish Application Server 3.1 Servlet 3.0//EN" "http://glassfish.org/dtds/glassfish-web-app_3_0-1.dtd">
<glassfish-web-app>
    <parameter-encoding default-charset="UTF-8" />
    <context-root>/</context-root>
    <session-config>
        <session-manager persistence-type="hazelcast">
        <!--<session-manager persistence-type="replicated">-->
          <manager-properties>
            <property name="persistenceFrequency" value="web-method"/>
            <property name="relaxCacheVersionSemantics" value="true"/>
          </manager-properties>
          <store-properties>
            <property name="persistenceScope" value="session"/>
          </store-properties>
        </session-manager>
        <cookie-properties>
          <property name="cookieSecure" value="false"/>
        </cookie-properties>
    </session-config>
</glassfish-web-app>
```
4.2. web.xml (some parts were left out)
```
<web-app xmlns:xsi="w3.org/2001/XMLSchema-instance"
 xmlns="xmlns.jcp.org/xml/ns/javaee"
 xsi:schemaLocation="xmlns.jcp.org/xml/ns/javaee xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
 version="3.1">
  <distributable/>
  <env-entry>
    <env-entry-name>jsf/ClientSideSecretKey</env-entry-name>
    <env-entry-type>java.lang.String</env-entry-type>
    <env-entry-value>xxxxxxxxxxxxxxxxxxx</env-entry-value>
  </env-entry>
  <!-- setting save viewstate by hidden field by client -->
  <context-param>
    <param-name>javax.faces.STATE_SAVING_METHOD</param-name>
    <param-value>client</param-value>
  </context-param>
  <filter>
 <filter-name>LoginFilter</filter-name>
 <filter-class>com.common.filter.LoginFilter</filter-class>
  </filter>
  <filter-mapping>
 <filter-name>LoginFilter</filter-name>
 <url-pattern>*.xhtml</url-pattern>
 <dispatcher>FORWARD</dispatcher>
 <dispatcher>REQUEST</dispatcher>
 <dispatcher>INCLUDE</dispatcher>
 <dispatcher>ASYNC</dispatcher>
 <dispatcher>ERROR</dispatcher>
  </filter-mapping>
  <servlet>
    <servlet-name>Faces Servlet</servlet-name>
    <servlet-class>javax.faces.webapp.FacesServlet</servlet-class>
    <load-on-startup>1</load-on-startup>
  </servlet>
  <servlet-mapping>
    <servlet-name>Faces Servlet</servlet-name>
    <url-pattern>*.xhtml</url-pattern>
  </servlet-mapping>
  <session-config>
    <session-timeout>30</session-timeout>
    <tracking-mode>COOKIE</tracking-mode>
  </session-config>
</web-app>
```

**Comment by hn310** [comment]
## Sample
https://github.com/hn310/session-replication

## Step to reproduce
1. At node 1, access: .../faces/ViewScoped.xhtml
2. Click [Set value] button, value of field will be updated to "new value" and appear on the screen.
3. Turn off node 1 (or change the jessionid to force the next request go to Node 2).
4. Click [Reload page] button, value of field will be lost.

**Note**
.../faces/RequestScoped.xhtml and .../faces/SessionScoped.xhtml worked as expected.

**Comment by hn310** [comment]
I will try again with the latest version.
But could you give me something like documentation or changelog to prove your point.
I just want to know for sure.
Thanks in advance.

**Comment by lprimak** [comment]
I fixed this particular bug in JSF myself. Trust me

**Comment by smillidge** [comment]
163 is a very old version of Payara can you see if you can reproduce this on Payara 5. If you require support on older versions or backported fixes please purchase commercial support https://www.payara.fish/support

**Comment by hn310** [comment]
Hi lprimak,
Do you remember exactly at which version this bug was fixed?
In case you don't remember, is the bug already fixed at 4.1.2.181?
We are afraid that moving from version 4 -> version 5 is some kind of "big change", and some parts of our code may not be compatible.
